internal/controllers/clustertemplate_controller.go

```diff
@@ -239,6 +252,71 @@ func validateConfigmapReference[T any](
 	return nil
 }
 
+// validateName return true if the ClusterTemplate name is the
+// format: <name>.<version> or if the cluster <name> is used in
+// another namespace for a ClusterTemplate, false otherwise
+func validateName(c client.Client, name, version, metadataName, namespace string) error {
+	if metadataName != name+"."+version {
+		return utils.NewInputError("failed to validate ClusterTemplate name %s, should be in the format <spec.name>.<spec.version>: %s", metadataName, name+"."+version)
+	}
+
+	allClusterTemplates := &oranv1alpha1.ClusterTemplateList{}
+	err := c.List(context.Background(), allClusterTemplates)
+	if err != nil {
+		return fmt.Errorf("could not get list of ClusterTemplate across the cluster: %w", err)
+	}
+
+	sameNameVersion := map[string]bool{}
+	sameMetadataName := map[string]bool{}
+	for _, aClusterTemplate := range allClusterTemplates.Items {
+		if aClusterTemplate.Namespace == namespace {
+			continue
+		}
+		if aClusterTemplate.Name == metadataName {
+			sameMetadataName[aClusterTemplate.Namespace] = true
+		}
+		if aClusterTemplate.Spec.Name == name &&
+			aClusterTemplate.Spec.Version == version {
+			sameNameVersion[aClusterTemplate.Namespace] = true
+		}
+	}
+	if len(sameMetadataName) != 0 {
+		return utils.NewInputError("failed to validate ClusterTemplate name %s, a identical name already exists in namespaces: %s",
+			metadataName, strings.Join(utils.MapKeysToSlice(sameMetadataName), ","))
+	}
+	if len(sameMetadataName) != 0 {
```

Comment:
Should this be  `if len(sameNameVersion) != 0` then? But I still feel this is redundant, it looks like you will never be able to reach to this point since it returns above if metadata.name is not same.

Reply:
Fixed. I agree, since the Xvalidation already rejects the ClusterTemplate with a invalid name, they won't exist in the cluster